sprocs/check_assessment_access.sql

```diff
@@ -38,7 +38,9 @@ BEGIN
                     END)
             ELSE 'None'
         END AS credit_date_string,
-        aar.time_limit_min,
+        CASE WHEN aar.time_limit_min IS NULL THEN NULL
+             ELSE LEAST(aar.time_limit_min, EXTRACT(EPOCH FROM aar.end_date - now() - INTERVAL '31 seconds') / 60)::integer
```

Comment:
Maybe we should add a comment here to explain why we are subtracting 31 seconds? (also, why 31 and not 30? looks suspicious! 😄)

Reply:
Suspicious indeed! 😆 Thing is -- `time_limit_min` only has minutes precision, so a 30s shift gives us 0 net change.

e.g.:
```
time_limit_min = 60m
now = 17:00
end = 18:00
```
If offset 30s, then no change: `end - now - 30s = 60m` (psql interval rounding)
If offset 31s, then change: `end - now - 31s = 59m`

So in practice, the buffer is 1 minute. Does that make sense or have I missed something?

Reply:
Ahhh!!! Excellent catch!!! I totally overlooked that.

Maybe change to 60 seconds in that case?

Reply:
> Maybe change to 60 seconds in that case?

So even though we subtract in 1-minute increments, I believe the exact offset still changes how quickly we steal that minute (higher offset = faster subtraction).

60s offset -> timer drops to 58m at 17:00:30
31s offset -> timer drops to 58m at 17:00:59

Reply:
Makes sense!

I'm totally fine to go with the 31-second solution here, but an alternative would be to have `check_assessment_access.sql` return `time_limit_sec` (`min` -> `sec`) so we can precisely encode `end_date - interval '60 seconds'`. This would require changing `studentAssessmentExam.ejs` to display seconds, and the assessment-instance creation code (`assessment.makeAssessmentInstance` and `assessment_instances_insert.sql`) to accept seconds rather than minutes. These are all small and easy changes, but I'm also fine to just stay with the 31-second solution.